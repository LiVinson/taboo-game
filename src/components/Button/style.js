import styled from "styled-components"

/*
  3 types of buttons:

  button
  button primary (extension of button)
  button-lg (extension of button) or button-lg primary (tbd)

*/

export const StyledButton = styled.button`
  border: none;
  color: ${(props) => props.theme.color.secondary};
  background-color: ${(props) => props.theme.color.offWhite};
  width: 47%;
  padding: 0.6rem 0rem;
  border-radius: 2px;
  font-size: 2.3rem;
  font-family: var(--font-family--display);
  text-transform: uppercase;

  &:active {
    outline: none;
    filter: brightness(0.8);
    transform: translateY(2px);
  }
`
export const StyledPrimaryButton = styled(StyledButton)`
  color: ${(props) => props.theme.color.offWhite};
  background-color: ${(props) => props.theme.color.secondary};
  width: 47%;
  &:active {
    filter: brightness(1.2);
  }
`
