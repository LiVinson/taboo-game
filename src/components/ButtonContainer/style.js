import styled from "styled-components"


export const StyledButtonContainer = styled.div`
  margin-top: 1.5rem;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
`