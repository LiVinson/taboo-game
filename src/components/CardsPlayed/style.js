import styled from "styled-components"

export const NoCardMessage = styled.p`
	font-size:2rem;
	color: ${props => props.theme.color.grayDark2};
`