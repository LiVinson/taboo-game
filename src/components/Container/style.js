import styled from 'styled-components'

export const StyledContainer = styled.div`
	position: relative;
	width: 100vw;

	min-height: 100vh;
	overflow: hidden;
	/* 
    Used for background image. Fills entire Container div. Before used so opacity does not impact children elements */
	&:before {
		display: block;
		content: '';
		height: 100%;
		width: 100%;
		background-image: url(/img/pattern-light.png);
		opacity: 0.6;
		position: absolute;
		top: 0;
		left: 0;
	}
`
