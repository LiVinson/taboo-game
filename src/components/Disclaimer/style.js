import styled from "styled-components"

export const StyledDisclaimer = styled.p`
    margin-top: 2.5rem;
    margin-bottom: .5rem;
    font-size: 1.4rem;
    color: ${props => props.theme.color.grayDark2};
    text-align: center;
    padding: 0 .5rem;
`