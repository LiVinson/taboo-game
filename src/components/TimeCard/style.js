import styled from 'styled-components'

//Container for the progress bar. Will be the background as the width of the inner bar decreases
export const ProgressBar = styled.div`
	width: 100%;
	position: relative;
	background-color: ${(props) => props.theme.color.grayDark1};
	height: 3rem;
	margin-bottom: 1rem;
	border-radius: 3px;
	border: solid 2px ${(props) => props.theme.color.grayDark2};
	box-shadow: 0px 3px 2px rgba(0, 0, 0, 0.3);
	display: flex;
	justify-content: space-between;
`

//Serves as the progress bar. Width is based on a percentage of time remaining / 60 seconds
export const Bar = styled.div`
	/* allows the color bar to overlap the numbers */
	position: absolute;
	top: 0;
	left: 0;
	width: ${(props) => props.width + '%'};
	/* If width is less than 16 (10 seconds or less remaining): change color */
	background-color: ${(props) =>
		parseInt(props.width) > 16 ? props.theme.color.accent3 : props.theme.color.accent2};
	height: 100%;
	z-index: 2;
`
export const TimeText = styled.div`
	/* positions numbers to the far right */
	position: absolute;
	/* when time = 0, centers time up message */
    right: ${(props) => (parseInt(props.width) > 0 ? 0 : '50%')};
	transform: ${(props) => {
		if (parseInt(props.width) > 0) {
			return 'translateX(0)'
		} else {
			return 'translateX(50%)'
		}
	}};
	padding: 0 0.5rem;
	height: 100%;
	font-size: 1.8rem;
	z-index: 5;
	color: ${(props) => props.theme.color.lightGray1};
`
