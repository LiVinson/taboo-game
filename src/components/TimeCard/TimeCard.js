import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import { ProgressBar, Bar, TimeText } from './style'

class TimeCard extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			timeRemaining: 60,
		}
	}

	//Calculates the difference between current time and time round ends in seconds
	//Updates state with the amount of time remaining every second until 0, then giver
	//calld method to endRound and change to PostRound.
	componentDidMount() {
		const { roundEndTime } = this.props
		let timeRemaining = moment(roundEndTime).diff(moment(), 'second')
		this.intervalId = setInterval(() => {
			if (timeRemaining >= 0) {
				timeRemaining--
				this.setState({
					timeRemaining,
				})
			} else {
				clearInterval(this.intervalId)
				//Only one player needs to update the round for everyone
				if (this.props.role === 'giver') {
					this.props.endRound()
				}
			}
		}, 1000)
	}

	componentWillUnmount() {
		if (this.intervalId) {
			clearInterval(this.intervalId)
		}
	}
	
	render() {
		const width = Math.round((this.state.timeRemaining / 60) * 100)
		return (
			<ProgressBar>
				<Bar width={width}></Bar>
				<TimeText width={width}>
					{this.state.timeRemaining > 0 ? this.state.timeRemaining : "Time's Up!"}
				</TimeText>
			</ProgressBar>
		)
	}
}

TimeCard.propTypes = {
	roundEndTime: PropTypes.number,
	role: PropTypes.string.isRequired,
	endRound: PropTypes.func.isRequired,
}
export default TimeCard
