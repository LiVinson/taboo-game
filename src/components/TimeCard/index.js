import TimeCard from "./TimeCard"

export default TimeCard