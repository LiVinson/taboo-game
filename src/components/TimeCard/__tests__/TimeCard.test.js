import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import TimeCard from '../TimeCard'
import { TimeText } from '../style'

describe('TimeCard rendering and functionality', () => {
	let defaultProps

	beforeEach(() => {
		jest.useFakeTimers()
		const currentTime = new Date()
		const endTime = currentTime.setTime(currentTime.getTime() + 60500)
		defaultProps = {
			roundEndTime: endTime,
			role: 'giver',
			endRound: jest.fn(),
		}
	})

	test('TimeCard renders correctly', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<TimeCard {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('Displays "Time is Up" message once time reaches 0', () => {
		const props = {
			...defaultProps,
			role: 'watcher',
		}

		const wrapper = shallow(<TimeCard theme={theme} {...props} />)

		//Mocking the setInterval. Makes it complete the timer before checking expect
		jest.runAllTimers()
		expect(wrapper.find(TimeText).text()).toBe("Time's Up!")
	})

	test('calls endRound once timer reaches 0 for the giver', () => {
		const props = {
			...defaultProps,
		}

		const wrapper = shallow(<TimeCard theme={theme} {...props} />)

        expect(props.endRound).toHaveBeenCalledTimes(0)
		//Mocking the setInterval. Makes it complete the timer before checking expect
        jest.runAllTimers()
        expect(props.endRound).toHaveBeenCalledTimes(1)
        

    })

	test('does not call endRound once timer reaches 0 for player not the giver', () => {
		const props = {
            ...defaultProps,
            role:"giverTeam"
		}

		const wrapper = shallow(<TimeCard theme={theme} {...props} />)

        expect(props.endRound).toHaveBeenCalledTimes(0)
		//Mocking the setInterval. Makes it complete the timer before checking expect
        jest.runAllTimers()
        expect(props.endRound).toHaveBeenCalledTimes(0)

    })
})
