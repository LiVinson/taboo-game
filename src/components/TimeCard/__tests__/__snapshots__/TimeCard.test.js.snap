// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeCard rendering and functionality TimeCard renders correctly 1`] = `
<div
  className="sc-AxjAm gtWhgV"
>
  <div
    className="sc-AxirZ lkJSxu"
    width={100}
  />
  <div
    className="sc-AxiKw hKElRV"
    width={100}
  >
    60
  </div>
</div>
`;
